csrc/ops/arith.cpp

```diff
@@ -1570,6 +1570,50 @@ std::vector<Val*> tensor_sizes(TensorView* inp) {
   return sizes;
 }
 
+std::vector<Val*> shape(TensorView* inp) {
+  auto iter_domains = TensorDomain::noReductions(inp->getMaybeRFactorDomain());
+  std::vector<Val*> shape(iter_domains.size(), nullptr);
+
+  for (auto idx : c10::irange(iter_domains.size())) {
+    shape[idx] = iter_domains[idx]->getMaybeExpandedExtent();
+  }
```

Comment:
Style suggestion. Feel free to ignore:
```suggestion
  std::vector<Val*> shape;
  shape.reserve(iter_domains.size());
  for (auto id : iter_domains) {
    shape.push_back(id->getMaybeExpandedExtent());
  }
```